lib/providers/httpTransaction/index.js

```diff
@@ -161,13 +217,26 @@ HttpTransaction.prototype.onClientReceive = function (data) {
     data: data.err,
     statusCode: data.statusCode
   })
+
+  var hasServerReceive = trace.events.some(function (event) {
+    return event.type === HttpTransaction.SERVER_RECV
+  })
+
+  var hasClientSend = trace.events.some(function (event) {
+    return event.type === HttpTransaction.CLIENT_SEND
+  })
+
+  // for now worker requests are not supported, clean them up
+  if (!hasServerReceive && hasClientSend) {
+    delete this.partials[data.id]
```

Comment:
shouldn't we also add a timeout which checks for size and emergency empty it with log messages?